modules/core/src/test/java/org/apache/ignite/internal/processors/cache/consistency/ExplicitTransactionalCacheConsistencyTest.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache.consistency;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.transactions.Transaction;
+import org.apache.ignite.transactions.TransactionConcurrency;
+import org.apache.ignite.transactions.TransactionIsolation;
+import org.apache.ignite.transactions.TransactionRollbackException;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+/**
+ *
+ */
+@RunWith(Parameterized.class)
+public class ExplicitTransactionalCacheConsistencyTest extends AbstractFullSetCacheConsistencyTest {
```

Comment:
Let's also check other cache methods in `readRepair` mode:

`IgniteCache#getAndPut`
`IgniteCache#getAndPutIfAbsent`
`IgniteCache#getAndRemove`
`IgniteCache#getAndReplace`
`IgniteCache#replace`
`IgniteCache#invoke`
`IgniteCache#invokeAll`

Reply:
Get before put is a possible extension of this feature.
I'll keep it in mind.